include/z64player.h

```diff
@@ -596,6 +596,14 @@ typedef enum {
     /* 4 */ PLAYER_LEDGE_CLIMB_4
 } PlayerLedgeClimbType;
 
+typedef enum {
+    /* -1 */ PLAYER_STICK_DIR_NONE = -1,
+    /*  0 */ PLAYER_STICK_DIR_UP,
+    /*  1 */ PLAYER_STICK_DIR_LEFT,
+    /*  2 */ PLAYER_STICK_DIR_DOWN,
+    /*  3 */ PLAYER_STICK_DIR_RIGHT
```

Comment:
what do you think about forward/backward instead of up/down

When looking at the control stick it's basically the same meaning, but when looking at player movement forward/backward is much more appropriate

Reply:
I should also say I'm not super convinced by this enum, it's for sure better readability for some parts but not all, I feel pretty confident about the og code just having used integers and counting them as "(closest) quarter turn units"

Reply:
alright I have mixed feelings about it but let's go with the enum

Reply:
Sorry for the churn but I think up/down is much more common, at least in speedrunning:

* ess up
* hess with ess down-left
* downroll
* jumpslash and hold up
* flick up-left

all of those sound awkward with forward/backward instead

Reply:
Those are all "absolute" directions, not relative to which way Link is facing. I guess I look at the control stick "from the top" instead of "from the side" when referring to the directions

Reply:
The enum values are relative to Link's yaw

For example as fig put it before, if Link is appearing looking right on the screen, then moving the stick up will be the PLAYER_STICK_LEFT value (left from Link)

Reply:
Oh I see, my bad

Reply:
Yeah it is relative to player. If you hold target and move around, the values work out like you would intuitively expect. When untargeted however, if you held right on the control stick while link was facing forward, it would register as DIR_RIGHT for one frame, and then when link is facing to the right, it will change to DIR_FORWARD even though youre holding right on the stick.

If theres any way to make this concept more clear in the docs somewhere id be down to add it. All I have done so far is say this in the header next to the array: `// Stores the control stick direction (relative to shape yaw) for the last 4 frames.`